web/src/main/java/se/inera/intyg/webcert/web/integration/interactions/listcertificatesforcarewithqa/ListCertificatesForCareWithQAResponderImpl.java

```diff
@@ -21,39 +21,47 @@
+import lombok.extern.slf4j.Slf4j;
 import org.apache.cxf.annotations.SchemaValidation;
 import se.inera.intyg.schemas.contract.Personnummer;
-import se.inera.intyg.webcert.web.csintegration.patient.GetPatientCertificatesWithQAFromCertificateService;
+import se.inera.intyg.schemas.contract.util.HashUtility;
+import se.inera.intyg.webcert.persistence.handelse.model.Handelse;
+import se.inera.intyg.webcert.web.csintegration.patient.GetCertificatesWithQAFromCertificateService;
 import se.inera.intyg.webcert.web.service.intyg.IntygService;
 import se.inera.intyg.webcert.web.service.intyg.dto.IntygWithNotificationsRequest;
 import se.inera.intyg.webcert.web.service.intyg.dto.IntygWithNotificationsResponse;
+import se.inera.intyg.webcert.web.service.notification.NotificationService;
 import se.riv.clinicalprocess.healthcond.certificate.listCertificatesForCareWithQA.v3.HandelseList;
 import se.riv.clinicalprocess.healthcond.certificate.listCertificatesForCareWithQA.v3.List;
 import se.riv.clinicalprocess.healthcond.certificate.listCertificatesForCareWithQA.v3.ListCertificatesForCareWithQAResponderInterface;
 import se.riv.clinicalprocess.healthcond.certificate.listCertificatesForCareWithQA.v3.ListCertificatesForCareWithQAResponseType;
 import se.riv.clinicalprocess.healthcond.certificate.listCertificatesForCareWithQA.v3.ListCertificatesForCareWithQAType;
 import se.riv.clinicalprocess.healthcond.certificate.listCertificatesForCareWithQA.v3.ListItem;
 import se.riv.clinicalprocess.healthcond.certificate.types.v3.HsaId;
-import se.riv.clinicalprocess.healthcond.certificate.types.v3.IIType;
-import se.riv.clinicalprocess.healthcond.certificate.v3.Intyg;
 
+@Slf4j
 @SchemaValidation
 public class ListCertificatesForCareWithQAResponderImpl implements ListCertificatesForCareWithQAResponderInterface {
 
     private final IntygService intygService;
-    private final GetPatientCertificatesWithQAFromCertificateService getPatientCertificatesWithQAFromCertificateService;
+    private final GetCertificatesWithQAFromCertificateService getCertificatesWithQAFromCertificateService;
+    private final NotificationService notificationService;
 
     public ListCertificatesForCareWithQAResponderImpl(IntygService intygService,
-        GetPatientCertificatesWithQAFromCertificateService getPatientCertificatesWithQAFromCertificateService) {
+        GetCertificatesWithQAFromCertificateService getCertificatesWithQAFromCertificateService,
+        NotificationService notificationService) {
         this.intygService = intygService;
-        this.getPatientCertificatesWithQAFromCertificateService = getPatientCertificatesWithQAFromCertificateService;
+        this.getCertificatesWithQAFromCertificateService = getCertificatesWithQAFromCertificateService;
+        this.notificationService = notificationService;
     }
 
     @Override
     public ListCertificatesForCareWithQAResponseType listCertificatesForCareWithQA(String s, ListCertificatesForCareWithQAType request) {
```

Comment:
The method has a lot going for it. Maybe you could refactor the actual work to a private method and wrap it with a try / finally and do the logging in the finally clause (consider that exceptions can occur so response.getList() could be null.